presto-iceberg/src/main/java/com/facebook/presto/iceberg/nessie/NessieConfig.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg.nessie;
+
+import com.facebook.airlift.configuration.Config;
+import com.facebook.airlift.configuration.ConfigDescription;
+
+import javax.validation.constraints.NotNull;
+
+import java.util.Optional;
+
+public class NessieConfig
+{
+    private String defaultReferenceName = "main";
+    private String serverUri;
+    private AuthenticationType authenticationType;
+    private String username;
+    private String password;
+    private String bearerToken;
+    private Integer readTimeoutMillis;
+    private Integer connectTimeoutMillis;
+    private String clientBuilderImpl;
+    private boolean compressionEnabled = true;
+
+    @NotNull
+    public String getDefaultReferenceName()
+    {
+        return defaultReferenceName;
+    }
+
+    @Config("iceberg.nessie.ref")
+    @ConfigDescription("The default Nessie reference to work on")
+    public NessieConfig setDefaultReferenceName(String defaultReferenceName)
+    {
+        this.defaultReferenceName = defaultReferenceName;
```

Comment:
Should we have isNullOrEmpty check here?

Reply:
I don't see this kind of validation being done in other Config classes, so I was assuming that the `@NotNull` annotation on `getDefaultReferenceName()` would take care of this.

Reply:
I think we do have null check examples, `StaticCatalogStoreConfig`

Reply:
added a null/empty check + added a test for it

Reply:
Just for compleness: I went over the config code to understand when things are being validated. I used the below test and temporarily removed the null/empty check in `setDefaultReferenceName`. Note that we provide an empty string to `iceberg.nessie.ref` in this example. 

```
@Test
public void testNessieMetastore()
{
    getOnlyElement(new IcebergPlugin().getConnectorFactories()).create(
                    "test",
                    ImmutableMap.of(
                            "hive.metastore.uri", "thrift://foo:1234",
                            "iceberg.catalog.type", "nessie",
                            "iceberg.nessie.ref", ""),
                    new TestingConnectorContext())
            .shutdown();
}
```

The validation of the config basically happens when the Plugin is being bootstrapped [here](https://github.com/prestodb/presto/blob/99748b79827022e9633103ee8948cd48b466a241/presto-iceberg/src/main/java/com/facebook/presto/iceberg/InternalIcebergConnectorFactory.java#L83-L86) and this is when the validation annotations (`@NotNull` / `@NotEmpty`/...) are being checked/verified. The startup of the plugin then correctly failed with the following exception:

```
com.google.inject.CreationException: Unable to create injector, see the following errors:

1) Error: Invalid configuration property iceberg.nessie.ref: must not be null or empty (for class com.facebook.presto.iceberg.nessie.NessieConfig.defaultReferenceName)

1 error

	at com.google.inject.internal.Errors.throwCreationExceptionIfErrorsExist(Errors.java:543)
	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:159)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:106)
	at com.google.inject.Guice.createInjector(Guice.java:87)
	at com.facebook.airlift.bootstrap.Bootstrap.initialize(Bootstrap.java:257)
	at com.facebook.presto.iceberg.InternalIcebergConnectorFactory.createConnector(InternalIcebergConnectorFactory.java:86)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.facebook.presto.iceberg.IcebergConnectorFactory.create(IcebergConnectorFactory.java:49)
	at com.facebook.presto.iceberg.TestIcebergPlugin.testNessieMetastore(TestIcebergPlugin.java:42)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at org.testng.TestRunner.privateRun(TestRunner.java:756)
	at org.testng.TestRunner.run(TestRunner.java:610)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:387)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:382)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:340)
	at org.testng.SuiteRunner.run(SuiteRunner.java:289)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
```
This would explain why the config classes don't perform validation checks in the setters but completely rely on the Validation annotations on the getters.

Reply:
Thanks for the fact check on the config validation!! Then looks like the null check could actually be safely removed since the getter would perform the null check?

Reply:
yep I removed the null check now